File: 1430-check-if-a-string-is-a-valid-sequence-from-root-to-leaves-path-in-a-binary-tree/1430-check-if-a-string-is-a-valid-sequence-from-root-to-leaves-path-in-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isValid(TreeNode* node,vector<int>& arr,int idx) {
        if(!node || idx == arr.size()) return false;
        if(node->val != arr[idx]) return false;
        if(!node->left && !node->right && idx == arr.size()-1) return true;
        return isValid(node->left,arr,idx+1) || isValid(node->right,arr,idx+1);
    }
    bool isValidSequence(TreeNode* root, vector<int>& arr) {
        return isValid(root,arr,0);
    }
};

File: 1430-check-if-a-string-is-a-valid-sequence-from-root-to-leaves-path-in-a-binary-tree/1430-check-if-a-string-is-a-valid-sequence-from-root-to-leaves-path-in-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    ArrayList<String> list = new ArrayList<>();
    public void getSeq(TreeNode node,String psf) {
        if(node == null) {
            return;
        }
        psf += node.val;
        if(node.left==null && node.right==null) {
            list.add(psf);
            return;
        }
        getSeq(node.left,psf);
        getSeq(node.right,psf);
    }
    public boolean isValidSequence(TreeNode root, int[] arr) {
         getSeq(root,"");
        // for(String s : list) System.out.print(s+" ");
        String check = "";
        for(int i=0;i<arr.length;i++) check+= String.valueOf(arr[i]);
            
        for(int i=0;i<list.size();i++) {
            if(list.get(i).equals(check)) return true;
        }
        return false;
    }
}